libraries/DataFlash/LogFile.cpp

```diff
@@ -710,7 +710,12 @@ void DataFlash_Class::Log_Write_GPS(const AP_GPS &gps, uint8_t i, uint64_t time_
     if (time_us == 0) {
         time_us = AP_HAL::micros64();
     }
-    const struct Location &loc = gps.location(i);
+    struct Location loc;
+    if (i >= gps.num_sensors()) {
+        loc = gps.location();
```

Comment:
This seems really messy. Why not have gps.location(3) manage the blended instance directly within the GPS library.